### DISCUSSION TITLE
Reading uart0 using os.dupterm(None, 1) - issues when going back to REPL when RX pin high

### DISCUSSION THREAD

**Discussion by Fisherman48** [micropython/micropython#10626] [ESP8266] [answered]
I want to read an uart 2k byte signal to esp8266 D1 mini. The only available is uart (0, 9600) but during development I have to use Thonny that using TX/RX on uart(0, 115200) for REPL.

Using Thonny 3.3.13
My D1 mini refuse to reconnect to REPL whith RX pin still high (no signal on uart).
I have to manually  or by switch to connect RX to uart signal source an disconnect before going back to REPL. 

I thought REPL was connected internal to RX /TX pins and not depending on RX pin status - but it looks that RX pin has to be free when going from REPL back again.

Is there a work around - not have to manually connect / disconnect?

Br Stefan

**Comment by robert-hh** [comment, marked as answer]
The inactive state of a UART line is high. Using os.dupterm() you can indeed detach and attach UART from REPL. But  in any case you 
- have to manage the UART configuration, especially setting the proper baud rate,  and you 
- have to physically detach other connections to RX and TX when using UART trough the built-in USB/UART bridge of the Wemos D1 mini. Otherwise the RX pin will be blocked. That's a hardware issue, not related to the ESP8266. To make that switchable, you will need some kind of external control of the signal.

There is a PR #7784 for MicroPython which provides support for a ESP8266.
But using that requires, that you build your own firmware. And this PR was not updated a while.

**Reply by Fisherman48** [reply]
Ok, its an hw issue - so I have to connect the TX from the source to my esp RX through a 3 state buffer (I had a 74HC244 in the IC box). So put the buffer in 'hi impedance state' duringbefore entering end after leaving the REPL session will work.
Thanks robert-hh for clearing this out
 /Fisherman
